From 7b58aeab24e5aa09cc0efadb87835f8f209a2436 Mon Sep 17 00:00:00 2001
From: Lim Siew Hoon <siew.hoon.lim@intel.com>
Date: Wed, 29 Jul 2020 02:28:24 +0800
Subject: [PATCH 1/2] gst/gstpluginloader.c: when env var is set do not fall 
 through to system plugin scanner

From c5c95f6062e4effe3806e399bf5a97cadd5238b6 Mon Sep 17 00:00:00 2001
From: dnagasun <devamekalai.nagasundaram@intel.com>
Date: Wed, 26 Feb 2020 06:18:55 +0800
Subject: [PATCH] gst/gstpluginloader.c: when env var is set do not fall
 through to system plugin scanner

From a1d7c582392c8bc87fa9411af77b20e011944357 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Thu, 25 Jan 2018 17:55:02 +0200
Subject: [PATCH] gst/gstpluginloader.c: when env var is set do not fall
 through to system plugin scanner

If we set a custom GST_PLUGIN_SCANNER env var, then we probably want to use that and only that.

Falling through to the one installed on the system is problamatic in cross-compilation
environemnts, regardless of whether one pointed to by the env var succeeded or failed.

Upstream-Status: Pending
Signed-offwq!: Alexander Kanavin <alex.kanavin@gmail.com>

Signed-off-by: dnagasun <devamekalai.nagasundaram@intel.com>
Signed-off-by: Lim Siew Hoon <siew.hoon.lim@intel.com>
---
 gst/gstpluginloader.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/gst/gstpluginloader.c b/gst/gstpluginloader.c
index d1e404d98..518486e55 100644
--- a/gst/gstpluginloader.c
+++ b/gst/gstpluginloader.c
@@ -475,9 +475,7 @@ gst_plugin_loader_spawn (GstPluginLoader * loader)
     helper_bin = g_strdup (env);
     res = gst_plugin_loader_try_helper (loader, helper_bin);
     g_free (helper_bin);
-  }
-
-  if (!res) {
+   } else {
     GST_LOG ("Trying installed plugin scanner");
 
 #ifdef G_OS_WIN32
-- 
2.17.1

